support/cas-server-support-oidc-core-api/src/main/java/org/apereo/cas/oidc/profile/OidcUserProfileViewRenderer.java

```diff
@@ -78,4 +90,11 @@ protected ResponseEntity renderProfileForModel(final Map<String, Object> userPro
         }
         return new ResponseEntity<>("Unable to produce user profile", HttpStatus.BAD_REQUEST);
     }
+
+    private Object useSingleValueForSingletonList(final Object value) {
+        if (value instanceof List && ((List) value).size() == 1) {
```

Comment:
In principal, this looks OK. However, to put a little more safety in:

1. You should check for Collection instead of List.
2. The method should be marked as static
3. I also wonder if you could enlist a method from CollectionUtils to calculate the single entry.